nilearn/plotting/surf_plotting.py

```diff
@@ -279,7 +279,13 @@ def plot_surf(surf_mesh, surf_map=None, bg_map=None,
 
         # treshold if indicated
         if threshold is None:
-            kept_indices = np.arange(surf_map_faces.shape[0])
+            # If no thresholding and nans, filter them out
+            if np.any(np.isnan(surf_map_faces)):
+                kept_indices = np.where(
+                                np.logical_not(
+                                    np.isnan(surf_map_faces)))[0]
+            else:
+                kept_indices = np.arange(surf_map_faces.shape[0])
```

Comment:
Is the if/else there for performance? I think the results would be equivalent if you just had:

```python
kept_indices = np.where(
    np.logical_not(
        np.isnan(surf_map_faces)))[0]
```

Reply:
Totally! I changed it in 914af1d
Thanks! :+1: